consensus/istanbul/validator/weighted.go

```diff
@@ -822,6 +827,13 @@ func calcTotalAmount(weightedValidators []*weightedValidator, stakingInfo *rewar
 	return totalStaking, gini
 }
 
+// setZeroWeight makes each validator's weight to zero
+func setZeroWeight(weightedValidators []*weightedValidator) {
+	for _, weightedVal := range weightedValidators {
+		atomic.StoreUint64(&weightedVal.weight, 0)
```

Comment:
@mckim19 Is it okay to set the weight as zero? because if weight zero, it seems that the validator allocates zero slot.. isn't it?
```
	for index, val := range valSet.validators {
		weight := val.Weight()
		for i := uint64(0); i < weight; i++ {
			candidateValsIdx = append(candidateValsIdx, index)
		}
	}
```

Reply:
If validators have zero weight, the proposers size is the same as the validators size.
Thank you!